packages/api/src/logger/index.ts

```diff
@@ -0,0 +1,194 @@
+ *
+ * Typically if logging to a transport stream or in production
+ * logs will not be prettified.
+ *
+ * In fact, the pino developers note:
+ * "We recommend against using pino-pretty in production,
+ * and highly recommend installing pino-pretty as a development dependency."
+ * See: https://github.com/pinojs/pino-pretty#programmatic-integration
+ *
+ * One exception to this rule may be Netlify functions logging.
+ * Its function logging output readability can benefeit from pretty-printing.
+ */
+export const isPretty = isDevelopment
+
+/**
+ * Defines the necessary pretty printing dependency
+ */
+export const prettifier = require('pino-pretty')
+
+/*
+ * List of keys to redact from log
+ *
+ * As an array, the redact option specifies paths that should have their values redacted from any log output.
+ *
+ * Each path must be a string using a syntax which corresponds to JavaScript dot and bracket notation.
+ *
+ * If an object is supplied, three options can be specified:
+ *
+ *      paths (String[]): Required. An array of paths
+ *      censor (String): Optional. A value to overwrite key which are to be redacted. Default: '[Redacted]'
+ *      remove (Boolean): Optional. Instead of censoring the value, remove both the key and the value. Default: false
+ */
+export const redactionsList: string[] | redactOptions = [
+  'host',
+  'email',
+  'accessToken',
+  'access_token',
+  'secret',
+]
+
+/**
+ * Determines if log level based on envars and
+ * development or deployment environment defaults
+ *
+ * Set `LOG_LEVEL` env to the desired logging level. In order of priority, available levels are:
+ *
+ * - 'fatal'
+ * - 'error'
+ * - 'warn'
+ * - 'info'
+ * - 'debug'
+ * - 'trace'
+ *
+ * The logging level is a __minimum__ level. For instance if `logger.level` is `'info'` then all `'fatal'`, `'error'`, `'warn'`,
+ * and `'info'` logs will be enabled.
+ *
+ * You can pass `'silent'` to disable logging.
+ *
+ */
+export const logLevel: LevelWithSilent | string =
+  typeof process.env.LOG_LEVEL !== 'undefined'
+    ? process.env.LOG_LEVEL
+    : isProduction
+    ? 'warn'
+    : 'trace'
+
+/**
+ * Defines an opinionated base logger configration how how to log and what to ignore.
+ *
+ * Defaults are:
+ *
+ * Colorize output
+ * Ingore some event attributes like hostname
+ * Prefix the log output with log level
+ * Use a shorted log message that omits server name
+ * Humanize time in GMT
+ * Set the default log level in dev or test to trace and warn in prod *or set via LOG_LEVEL)
+ * Nest objects under an `api` key to avoid conflicts with pino properties
+ * Redact the host
+ *
+ * See: https://github.com/pinojs/pino/blob/master/docs/api.md
+ *      https://github.com/pinojs/pino-pretty
+ */
+
+export const defaultLoggerOptions: LoggerOptions = {
+  prettyPrint: isPretty && {
+    colorize: true,
+    ignore: 'hostname,pid',
+    levelFirst: true,
+    messageFormat: false,
+    translateTime: true,
+  },
+  prettifier: isPretty && prettifier,
```

Comment:
Is it OK for this to be false?

Reply:
Looks like it is:

```js
function getPrettyStream (opts, prettifier, dest, instance) {
  if (prettifier && typeof prettifier === 'function') {
    prettifier = prettifier.bind(instance)
```

FYI - This ensures that the dependency is loaded. If not the function may not include.